packages/server/src/fhir/search.ts

```diff
@@ -218,14 +228,25 @@ interface GetSelectQueryForSearchOptions extends GetBaseSelectQueryOptions {
 }
 function getSelectQueryForSearch<T extends Resource>(
   repo: Repository,
-  searchRequest: SearchRequest<T>,
+  searchRequest: SearchRequestWithCountAndOffset<T>,
   options?: GetSelectQueryForSearchOptions
 ): SelectQuery {
   const builder = getBaseSelectQuery(repo, searchRequest, options);
   addSortRules(builder, searchRequest);
-  const count = searchRequest.count as number;
+  const count = searchRequest.count;
   builder.limit(count + (options?.limitModifier ?? 1)); // Request one extra to test if there are more results
-  builder.offset(searchRequest.offset ?? 0);
+  if (searchRequest.offset > 0) {
+    builder.offset(searchRequest.offset);
+  }
+  if (searchRequest.cursor) {
+    const cursor = parseCursor(searchRequest.cursor);
+    if (cursor) {
+      builder.orderBy(new Column(searchRequest.resourceType, 'lastUpdated', false));
+      if (cursor.instant) {
+        builder.where(new Column(searchRequest.resourceType, 'lastUpdated'), '>', cursor.instant);
```

Comment:
Using `>` seems like it has an edge case when two resources have the same `lastUpdated` but only one of them is included in the previous page/batch. The other resource would be skipped.

Reply:
New model:
* Sort by `lastUpdated` ascending and `id` asc
* Filter on `(lastUpdated=x and id>y) or lastUpdated>x`

After that, no edge cases with multiple resources having the same `lastUpdated`

✅